api/lql_test.go

```diff
@@ -0,0 +1,632 @@
+	"END_TIME_RANGE": "111213141516",
+	"QUERY_TEXT": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				StartTimeRange: "1970-01-01T00:11:18Z",
+				EndTimeRange:   "1973-07-11T04:32:21Z",
+				QueryText:      "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+				QueryBlob: `{
+	"START_TIME_RANGE": "678910",
+	"END_TIME_RANGE": "111213141516",
+	"QUERY_TEXT": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+		},
+		LQLQueryTest{
+			Name: "json-blob-lower",
+			Input: &api.LQLQuery{
+				QueryBlob: `{
+	"start_time_range": "678910",
+	"end_time_range": "111213141516",
+	"query_text": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				StartTimeRange: "1970-01-01T00:11:18Z",
+				EndTimeRange:   "1973-07-11T04:32:21Z",
+				QueryText:      "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+				QueryBlob: `{
+	"start_time_range": "678910",
+	"end_time_range": "111213141516",
+	"query_text": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+		},
+		LQLQueryTest{
+			Name: "lql-blob",
+			Input: &api.LQLQuery{
+				QueryBlob: "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				QueryText: "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+				QueryBlob: "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }",
+			},
+		},
+		LQLQueryTest{
+			Name: "overwrite-blob",
+			Input: &api.LQLQuery{
+				StartTimeRange: "0",
+				EndTimeRange:   "1",
+				QueryText:      "should not overwrite",
+				QueryBlob: `{
+	"START_TIME_RANGE": "678910",
+	"END_TIME_RANGE": "111213141516",
+	"QUERY_TEXT": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+			Return: nil,
+			Expected: &api.LQLQuery{
+				StartTimeRange: "1970-01-01T00:00:00Z",
+				EndTimeRange:   "1970-01-01T00:00:00Z",
+				QueryText:      "should not overwrite",
+				QueryBlob: `{
+	"START_TIME_RANGE": "678910",
+	"END_TIME_RANGE": "111213141516",
+	"QUERY_TEXT": "my_lql(CloudTrailRawEvents e) { SELECT INSERT_ID LIMIT 10 }"
+}`,
+			},
+		},
+	}
+)
+
+type LQLTranslateTimeTest struct {
+	Name       string
+	Input      string
+	ReturnTime string
+	ReturnErr  interface{}
+}
+
+type LQLValidateRangeTest struct {
+	Name       string
+	Input      api.LQLQuery
+	AllowEmpty bool
+	Return     interface{}
+}
+
+type LQLValidateTest struct {
+	Name   string
+	Input  *api.LQLQuery
+	Return interface{}
+}
+
+type LQLQueryTest struct {
+	Name     string
+	Input    *api.LQLQuery
+	Return   interface{}
+	Expected *api.LQLQuery
+}
+
+func TestTranslateTime(t *testing.T) {
+	for _, lqlTranslateTimeTest := range lqlTranslateTimeTests {
+		t.Run(lqlTranslateTimeTest.Name, func(t *testing.T) {
+			outTime, err := api.LQLQuery{}.TranslateTime(lqlTranslateTimeTest.Input)
+			if err == nil {
```

Comment:
Inside tests, we normally do `err == nil` or `err != nil`, instead we use:
```suggestion
			if assert.Nil(t, err) {
```

Reply:
I don't want to `assert.Nil` here, I simply want different assert logic when `err == nil`...